solr/core/src/test/org/apache/solr/core/FSPRSTest.java

```diff
@@ -41,6 +43,61 @@
 @LogLevel("org.apache.solr.common.cloud.PerReplicaStatesOps=DEBUG")
 public class FSPRSTest extends SolrCloudTestCase {
 
+  public void testShardSplitWithNodeset() throws Exception {
+    String COLL = "prs_shard_split_nodeset";
+    MiniSolrCloudCluster cluster =
+        configureCluster(2)
+            .withJettyConfig(jetty -> jetty.enableV2(true))
+            .addConfig("conf", configset("conf2"))
+            .configure();
+
+    try {
+      CollectionAdminRequest.createCollection(COLL, "conf", 2, 2)
+          .setMaxShardsPerNode(4)
+          .process(cluster.getSolrClient());
+      cluster.waitForActiveCollection(COLL, 2, 4);
+
+      JettySolrRunner jetty = cluster.startJettySolrRunner();
```

Comment:
I like adding an empty node to the cluster here; but I wonder rather than placing the new non-leader replicas (via `createNodeSet` always on the newly-created node, could we pick the node randomly? e.g.:
```java
String nodeName = pickRandom(cluster.getSolrClient().getClusterStateProvider().getLiveNodes().toArray(new String[0]));
```
This would require increasing maxShardsPerNode, but should be more robust against the possibility of some _other_ incidental cause of always placing the new non-leader replicas on the newly-created (initially empty) node.

Reply:
@noblepaul any thought on this!

Reply:
I have chosen a random jetty as the target for the second split @magibney

Reply:
Thanks, that looks good. I'd be inclined to randomize the createNodeSet param for the _first_ split command as well (if the idea is for `createNodeSet` to support setting an arbitrary node, then it stands to reason that the tests should pick the node arbitrarily); but I think it's good with the current change.